compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -750,8 +754,10 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
 
         if tcx.sess.meta_stats() {
             let mut zero_bytes = 0;
-            for e in self.opaque.data.iter() {
-                if *e == 0 {
+            self.opaque.file().seek(std::io::SeekFrom::Start(0)).unwrap();
```

Comment:
I added this `seek` to read the file from the start.

Reply:
There are writes to this file after this stat is computed.
Could you add an assertion below that we get back to the position we left?
Should we flush the encoder buffer before computing this stat?

Reply:
Fixed by https://github.com/rust-lang/rust/pull/96544/commits/5ed4574197badfad1b87ae9c9c4fa7d06a112d47 .

I add `flush` before rewinding the file. In addition, I store the position before rewinding and asserts after counting zero bytes.